src/net.c

```diff
@@ -991,6 +991,13 @@ static int hyper_setup_interface(struct rtnl_handle *rth,
 	req.n.nlmsg_type = RTM_NEWADDR;
 	req.ifa.ifa_family = AF_INET;
 
+	if (iface->device) {
```

Comment:
@dlespiau I agree, we do tend to check the same pointers at every call. I like your suggestion, I'll push a change for this.